### ISSUE TITLE
ResNet50 (class weights / 2nd fine tuning)

### ISSUE THREAD

**Issue by hanhee-cho** [pladata-encore/DA36-mini3-hairwego#17]
## 🌈 Success criteria
1. class weights
2. fine tuning 2단계 (learning rate 작게 -> 0.001)

## 👷 To-do
- [ ] class weights 계산 - 조정
```
# @title Class weights 적용

from sklearn.utils.class_weight import compute_class_weight
import numpy as np

# 클래스 라벨과 샘플 수를 기반으로 가중치 계산
class_weights = compute_class_weight(
    class_weight='balanced',
    classes=np.unique(tr_labels),
    y=tr_labels
)

# 딕셔너리 형태로 변환
class_weight_dict = dict(enumerate(class_weights))

print(class_weight_dict)
```
- [ ] Fine Tuning 2단계
```
# 특성추출층 unfreeze (Fine Tuning 2단계)
for layer in model.layers:
    if not isinstance(layer, layers.BatchNormalization):
        layer.trainable = True
```

### ResNet50.loss-0.67-accuracy-0.75.h5
![Image](https://github.com/user-attachments/assets/19522dbb-d30c-48e8-86a2-d034bb6e3d20)
